### PR TITLE
Add hash set check for not changed value.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: structure/hash.go [hunk 1]
```diff
@@ -105,8 +105,9 @@ func (t *TxStructure) updateHash(key []byte, field []byte, fn func(oldValue []by
 		return errors.Trace(err)
 	}
 
+	newMeta := meta
 	if oldValue == nil {
-		meta.Length++
+		newMeta.Length++
```
Comment (siddontang) [inline structure/hash.go]: if oldValue is nil and newValue is nil too, I don't think we need add length and can return directly.

Comment (qiuyesuifeng) [inline structure/hash.go]: Aggreed, we should add check for nil value.

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**